### PR TITLE
Rank mentors by number of mentees

### PR DESCRIPTION
**Pull request by tsleal29**

### CODE REVIEW

**Comment by tsleal29** [comment]
Screen Shot 2020-08-10 at 18 04 19
Screen Shot 2020-08-10 at 18 03 59

File: app/models/user.rb [hunk 1]
```diff
@@ -12,6 +12,10 @@ def mentor?
     !!User.find_by(mentor_id: self.id)
   end
 
+  def mentees_count
+    User.where(mentor: self.id).count
```
Comment (fwuensche) [inline app/models/user.rb:16]: You can add an association to user by using: 

```
has_many :mentees, class_name: 'User', foreign_key: :mentor_id 
```

File: app/controllers/mentors_controller.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ def index
     # TODO: order by ascending number of pairs (available mentors should be first)
     # TODO: count only active S&H mentorships
     # TODO: preload mentorship pairs to view (avoid n+1)
-    @mentors = User.where(id: users_with_mentor.select(:mentor_id))
+    @mentors = User.where(id: users_with_mentor.select(:mentor_id)).sort_by(&:mentees_count).reverse
```
Comment (fwuensche) [inline app/controllers/mentors_controller.rb:10]: There's a N+1 query here. We could let it pass for now but should probably fix in another PR.

File: app/views/mentors/index.slim [hunk 3]
```diff
@@ -12,8 +12,8 @@ table.table
...
     tr
       td = link_to mentor.name, mentor_path(mentor)
-      td
-      td
-      td
-      td
+      td = mentor.location
+      td = mentor.language
+      td = mentor.ruby_level
+      td = mentor.react_level
```
Comment (fwuensche) [inline app/views/mentors/index.slim:18]: these changes are not related to the PR, no?
